components/camel-test/camel-test-junit6/src/test/java/org/apache/camel/test/junit6/patterns/FilterCreateCamelContextPerClassTest.java

```diff
@@ -42,7 +44,9 @@ public void testSendMatchingMessage() throws Exception {
 
     @Test
     public void testSendNotMatchingMessage() throws Exception {
-        getMockEndpoint("mock:result").expectedMessageCount(0);
+        MockEndpoint me = getMockEndpoint("mock:result");
+        me.expectedMessageCount(0);
```

Comment:
It was failing here in the first run (as it had one exchange of the first test). The second execution is fine as maven rerun only this one individually.